src/transformers/models/whisper/generation_whisper.py

```diff
@@ -954,6 +964,25 @@ def _need_fallback(
                 should_skip = True
 
         return needs_fallback, should_skip
+    
+    def _expand_variables_for_generation(self, input_features, seek, max_frames, init_tokens, batch_size, condition_on_prev_tokens, generation_config): 
```

Comment:
Can you refer to `_expand_inputs_for_generation` to see how this can potentially be done in a cleaner way (especially for the torch tensors)?

https://github.com/huggingface/transformers/blob/38aa42d459ad5a904077ecf683fb5b670f4f2540/src/transformers/generation/utils.py#L597C9-L597C38